kite-morphlines/kite-morphlines-phonenumber/src/main/java/org/kitesdk/morphline/phonenumber/PhoneNumberBuilder.java

```diff
@@ -0,0 +1,166 @@
+				final MorphlineContext context) {
+			super(builder, config, parent, child, context);
+
+			this.phoneNumberUtil = PhoneNumberUtil.getInstance();
+			this.inputField = getConfigs().getString(config, FIELD_INPUT_FIELD);
+
+			final String outputFieldPrefix = getConfigs()
+					.getString(config, FIELD_OUTPUT_FIELD_PREFIX, DEFAULT_OUTPUT_FIELD_PREFIX);
+			this.outputPhoneField = outputFieldPrefix + SUFFIX_PHONE;
+			this.outputCountryCodeField = outputFieldPrefix + SUFFIX_COUNTRY_CODE;
+			this.outputRegionCodeField = outputFieldPrefix + SUFFIX_REGION_CODE;
+			this.regionCodeField = getConfigs().getString(config, FIELD_REGION_CODE_FIELD, null);
+
+			this.defaultRegionCode = getConfigs().getString(config, FIELD_DEFAULT_REGION_CODE, DEFAULT_REGION_CODE);
+			if (!this.phoneNumberUtil.getSupportedRegions().contains(this.defaultRegionCode)) {
+				throw new MorphlineCompilationException(
+						"The specified default region code is not valid: " + this.defaultRegionCode, config);
+			}
+
+			final String phoneNumberFormatString = getConfigs()
+					.getString(config, FIELD_OUTPUT_PHONE_FORMAT, DEFAULT_OUTPUT_PHONE_FORMAT)
+					.toUpperCase(Locale.ENGLISH);
+			try {
+				this.phoneNumberFormat = PhoneNumberUtil.PhoneNumberFormat.valueOf(phoneNumberFormatString);
+			} catch (IllegalArgumentException ex) {
+				throw new MorphlineCompilationException(
+						"The specified phone number format is not valid: " + phoneNumberFormatString, config);
+			}
+
+			validateArguments();
+		}
+
+		@Override
+		protected boolean doProcess(final Record record) {
+
+			final Object objValue = record.getFirstValue(inputField);
+			if (objValue == null) {
+				LOG.debug("Input field ({}) was null", inputField);
+				return false;
+			} else if (!(objValue instanceof String)) {
+				if (LOG.isWarnEnabled()) {
+					LOG.warn("Input field ({}) has type {} but String was expected", inputField,
+							objValue.getClass().getCanonicalName());
+				}
+				return false;
+			}
+			final String inputValue = (String) objValue;
+
+			final String regionCode = getRegionCode(record);
+
+			try {
+				final Phonenumber.PhoneNumber phoneNumber = phoneNumberUtil.parse(inputValue, regionCode);
+				final String formattedPhoneNumber = phoneNumberUtil.format(phoneNumber, phoneNumberFormat);
+				final int countryCode = phoneNumber.getCountryCode();
+				record.put(outputPhoneField, formattedPhoneNumber);
+				record.put(outputCountryCodeField, Integer.toString(countryCode));
+				record.put(outputRegionCodeField, phoneNumberUtil.getRegionCodesForCountryCode(countryCode).get(0));
+			} catch (NumberParseException ex) {
+				LOG.warn("Exception while parsing phone number: " + inputValue, ex);
+				return false;
+			}
+
+			return super.doProcess(record);
+		}
+
+		private String getRegionCode(final Record record) {
+			if (regionCodeField == null) {
+				return defaultRegionCode;
+			}
+			final Object objValue = record.getFirstValue(regionCodeField);
```

Comment:
if we iterate over all values in inputField (see comment above) then also consider iterating over all values here.